frontend/fingertips-frontend/components/molecules/Inequalities/BarChart/index.tsx

```diff
@@ -50,31 +46,43 @@ const generateInequalitiesBarChartTooltipList = (
 
 export function InequalitiesBarChart({
   barChartData,
-  dynamicKeys,
   yAxisLabel,
   measurementUnit,
-  benchmarkValue = undefined,
   type = InequalitiesTypes.Sex,
   areasSelected = [],
 }: Readonly<InequalitiesBarChartProps>) {
   const xAxisTitlePrefix = 'Inequality type:';
+  const { inequalities } = barChartData.data;
+  const { benchmarkValue, inequalityDimensions: barChartFields } =
+    getAggregatePointInfo(inequalities);
 
-  const barChartFields = mapToGetBarChartKeys[type](dynamicKeys);
+  // for sex inequality we always want Male, Female which is reverse alphabetical order
+  // pending a better solution where an order key is supplied by API
+  if (type === InequalitiesTypes.Sex) barChartFields.reverse();
 
   const yAxisMaxValue = getMaxValue([
-    ...barChartFields.map(
-      (field) => barChartData.data.inequalities[field]?.value
-    ),
+    ...barChartFields.map((field) => inequalities[field]?.value),
     benchmarkValue,
   ]);
 
   const seriesData: Highcharts.SeriesOptionsType[] = [
     {
       type: 'bar',
-      data: barChartFields.map((field) => ({
-        name: field,
-        y: barChartData.data.inequalities[field]?.value,
-      })),
+      data: barChartFields.map((field) => {
+        const colours = getBenchmarkTagStyle(
+          BenchmarkLabelGroupType.RAG,
+          inequalities[field]?.benchmarkComparison
+            ?.outcome as BenchmarkLabelType
+        );
+        const backgroundColor = colours?.backgroundColor ?? undefined;
```

Comment:
Isn't the nullish coalescing operator unnecessary here, given you're using optional chaining?
```suggestion
        const backgroundColor = colours?.backgroundColor;
```